### DISCUSSION TITLE
Import markers with different point colors

### DISCUSSION THREAD

**Discussion by JoseIber** [organicmaps/organicmaps#8717] [Q&A]
Hello community. I have a question. Is it possible to import a KML with points with different colors based on a label? They can be edited once imported, but I would like to import it with that color data and not have to edit them one by one. Could it be? Thank you so much

**Comment by cyber-toad** [comment]
@JoseIber now color for points should be specified in kml/gpx to be imported in OrganicMaps. You can set color manually with GoogleEarth before import or enrich color values using some script to parse/modify your kml.

**Reply by JoseIber** [reply]
Hello cyber-toad. Thank you very much for your answer. The truth is that I have tried it in many ways, I have edited different scripts, but they don't work for me, they always come out red. I would like it to assign the color based on the value of the "routes" tag, but it seems that is not possible. This is part of my script:

icon_dict = {
    "Otra": 'placemark-bluegray',
    "Realizadas": 'placemark-pink',
    "Albert": 'placemark-gray',
    "Jose": 'placemark-yellow',
    "Ruth": 'placemark-blue',
    "Marta": 'placemark-green',
    "Pendientes": 'placemark-orange',
    "Interesantes": 'placemark-purple',
    "Por valorar": 'placemark-red',
    "No hacer": 'placemark-lima'
}

# Crear estilos para cada ícono
styles = {}
for stage, icon in icon_dict.items():
    style = simplekml.Style()
    style.iconstyle.icon.href = f'https://omaps.app/placemarks/{icon}.png'
    style.iconstyle.scale = 1.0
    styles[stage] = style

**Reply by cyber-toad** [reply]
@JoseIber  if you share your kml file I can try to check why colors are not applied.

**Reply by JoseIber** [reply]
Sure, thank you very much. Here I show you an example file.
[kml.txt](https://github.com/user-attachments/files/16389407/kml.txt)

**Reply by matkoniecz** [reply]
I use following styling and it works

```
<Placemark>
    <name>Poczta Polska</name>
    <styleUrl>#placemark-gray</styleUrl>
    <Point><coordinates>19.578827,50.2691</coordinates></Point>
  </Placemark>
```

Not sure why you do

```
  <IconStyle>
    <Icon>
      <href>https://omaps.app/placemarks/placemark-bluegray.png</href>
    </Icon>
  </IconStyle>
```

is something like that documented as supported?

**Reply by cyber-toad** [reply]
Yep, I also see that IconStyle is saved here kml::SaveStyle but don't see how it is parsed, so @matkoniecz example with styleUrl looks valid.

**Comment by JoseIber** [comment]
Thank you so much. I followed your instructions and it worked. Thanks for helping 🙌
